app/upgrades/README.md

```diff
@@ -99,7 +100,7 @@ func CreateUpgradeHandler(
 	configurator module.Configurator,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
-		vm[{moduleName}] = 2 // <- ADD THIS
+		vm[{moduleName}] = {new-consensus-version} // <- ADD THIS
```

Comment:
Its should be `old-consensus-version`. Because semantically, `vm` is `fromVersionMap`. If we leave it as defined, it contains old versions stored in state. And we defined `new-consensus-version` in `module.ConsensusVersion()`
For each difference between `old` and `new`, the module-specific store migrations are run and the respective consensus version of each upgraded module is incremented. So, if we set `vm[{moduleName}] = {new-consensus-version}` here, migration will not be run.

Reply:
In addition, we dont need to care about `vm` in `upgradeHandler`. Just increase version in `module.ConsensusVersion()`. `vm` already contained the old version in it and the migration is automatically executed

Reply:
Great catch! I'll update these docs once we finalize the approach